recording_server/src/media_context.ts

```
@@ -102,107 +186,213 @@ export class SoundContext extends MediaContext {
+        }
+        
+        args.push('-re', '-i', pathname)
+        
+        if (IS_MACOS) {
+            // macOS: Record audio (virtual devices are complex on macOS)
+            const outputFile = `/tmp/virtual_audio_${Date.now()}.wav`
+            args.push('-f', 'wav', outputFile)
+            console.log(`🍎 macOS: Recording audio to ${outputFile}`)
+        } else if (CONFIG.audioFormat === 'pulse') {
+            // Linux with PulseAudio (NixOS)
+            args.push('-f', CONFIG.audioFormat, '-acodec', CONFIG.audioCodec!, CONFIG.microDevice)
```

Biome (1.9.4)

[error] Forbidden non-null assertion.



(lint/style/noNonNullAssertion)